### ISSUE TITLE
Non-persistence of setting not to warn when closing multiple tabs

### ISSUE THREAD

**Issue by johanbluecreek** [piroor/treestyletab#2735]
## Short description

When I close a parent tab and the children are to be closed I get a warning that I'm closing multiple tabs and I have the option to ignore this warning in the future. When I close a parent the next time, I still get the warning even if I asked not to be warned.

## Steps to reproduce

This test is done on debian stable with latest version of firefox in their stable repositories (78.3.0esr/78.3.0esr-1~deb10u1)

 1. Start Firefox with clean profile.
 2. Install TST.
 3. Open a tab
 4. Spawn children to that tab
 5. Collapse the branch to hide the children
 6. Close the parent
 7. Get warning about closing multiple tabs and uncheck "Warn me when I attempt to close multiple tabs"
 8. Press "Close tabs"
 9. Repeat items 3-6.
 10. Get the warning again.

## Expected result

When taking the option to not be warned about closing multiple tabs I expect not to be warned when closing multiple tabs in the future.

## Actual result

I get warned about closing multiple tabs when I asked not to.

## Environment

 * Platform (OS): Debian stable
 * Version of Firefox: 78.3.0esr/78.3.0esr-1~deb10u1
 * Version (or revision) of Tree Style Tab: 3.5.31

**Comment by piroor** [comment]
What is the method did you try to close tabs? Did you tried both times with an exact same method?

TST provides two options "Warn me when I attempt to close multiple tabs" and "Warn me when I attempt to close multiple tabs by a normal click on a closebox" and both they are active by default. Even if the first option is deactivated, you'll see a confirmation for closing of multiple tabs with a click on the closebox.

**Comment by johanbluecreek** [comment]
> What is the method did you try to close tabs?

Clicking on the X on the parent in the sidebar tree.

> Did you tried both times with an exact same method?

I have tried variations of this method tens of times on two different machines with the same versions of firefox, TST, and latest updated Debian stable.

> TST provides two options "Warn me when I attempt to close multiple tabs" and "Warn me when I attempt to close multiple tabs by a normal click on a closebox" and both they are active by default. Even if the first option is deactivated, you'll see a confirmation for closing of multiple tabs with a click on the closebox.

I always get the dialog that says "Warn me when I attempt to close multiple tabs", even if I uncheck that box previously.

How do I trigger the "Warn me when I attempt to close multiple tabs by a normal click on a closebox" dialog? What is the "closebox"?

**Comment by piroor** [comment]
Thanks! I've realized that the checkbox state is not saved unexpectedly when you close tabs with the closebox ("X" button in tabs). The state was saved as expected only when the dialog was shown for closing of tabs via a middle click. The commit 030e773 should fix the problem.

**Comment by piroor** [comment]
By the way, the second "by a normal click on a closebox" option was separated because I accidentally closed collapsed tree too many times when I just aimed to switch tab focus.

**Comment by johanbluecreek** [comment]
I downloaded the nightly build of TST that appears to have that commit (by checking the latest entry in the 'CI/CD action'), loaded it temporarily as per the instructions in the read me, and it works! I spawned a bunch of children, closed parent and got the warning and opted out, redid it once more, and I got no warning. It is all in nightly, but since it is in the pipe to get rolled out I'm closing the issue.

Thank you!
